include/libcyphal/transport/errors.hpp

```diff
@@ -30,10 +34,17 @@ struct MemoryError final
 struct CapacityError final
 {};
 
-struct PlatformError final
+class IPlatformError
 {
-    std::uint32_t code{};
+protected:
+    IPlatformError()                                     = default;
+    ~IPlatformError()                                    = default;
+    IPlatformError(const IPlatformError&)                = default;
+    IPlatformError(IPlatformError&&) noexcept            = default;
+    IPlatformError& operator=(const IPlatformError&)     = default;
+    IPlatformError& operator=(IPlatformError&&) noexcept = default;
 };
+using PlatformError = ImplementationCell<IPlatformError, cetl::unbounded_variant<sizeof(void*) * 3>>;
```

Comment:
This `3` is magic - I guess we will adjust on-fly if some concrete implementation of `IPlatformError` doesn't fit anymore.

Reply:
Is there at least some basis for why 3 was chosen?

Reply:
Most simple lambdas (capturing f.e. `this`) will fit into 24 bytes.

Reply:
>Is there at least some basis for why 3 was chosen?

vibes-driven design